ksqldb-engine/src/test/java/io/confluent/ksql/integration/SecureIntegrationTest.java

```diff
@@ -243,6 +254,196 @@ public void shouldWorkWithMinimalPrefixedAcls() {
     assertCanAccessClusterConfig(prefix);
   }
 
+  @Test
+  public void shouldClassifyMissingSourceTopicExceptionAsUserError() {
+    // Given:
+    final String serviceId = "my-service-id_";  // Defaults to "default_"
```

Comment:
These new tests are hard to read since it's not clear to me which parts of the "givens" are relevant for the test and which are just boilerplate. Can we refactor the common parts/boilerplate into a helper method (e.g., `givenTestSetupWithAclsForQuery()`) so only the relevant parts are left in each test?